weave/logs.py

```diff
@@ -102,13 +102,18 @@ def default_log_filename() -> typing.Optional[str]:
     return get_logfile_path(f"{pid}.log")
 
 
-def env_log_level() -> typing.Any:
-    # Default to only showing ERROR logs, unless otherwise specified
-    default_level = "ERROR"
+def env_log_level() -> int:
+    level_str_from_env = os.environ.get("WEAVE_LOG_LEVEL")
+    if level_str_from_env:
+        level_int_from_env = logging.getLevelName(level_str_from_env)
+        if isinstance(level_int_from_env, int):
+            return level_int_from_env
+
     if os.environ.get("WEAVE_SERVER_ENABLE_LOGGING"):
-        default_level = "INFO"
-    level = os.environ.get("WEAVE_LOG_LEVEL", default_level)
-    return logging.getLevelName(level)
+        return logging.INFO
+
+    # Default to only showing ERROR logs
+    return logging.ERROR
```

Comment:
- The only time we ever need to convert from a string log level is when reading `WEAVE_LOG_LEVEL` env var
- Branching logic reorganized for additional intuitiveness